core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java

```diff
@@ -74,38 +81,18 @@ private void registerServices(ServiceExtensionContext context) {
         var agentService = new ParticipantAgentServiceImpl();
         context.registerService(ParticipantAgentService.class, agentService);
 
-        // Lazily load the contract offer framework into this core module since the implementation is provided by an extension that will require other core services
-        var cofSupplier = new CofSupplier(context, monitor);
-        var contractOfferService = new ContractOfferServiceImpl(agentService, cofSupplier, assetIndex);
+        var policyEngine = new PolicyEngineImpl();
```

Comment:
There should be on entry in the Set returned by `provides()` per registered service. Also, those Strings should be constants on the interface (`FEATURE`).

Reply:
Provides and Requires are per feature, not necessarily per service. It will become unmanageable if each extension has to declare every service it relies on. The intent of provides/requires was to declare this at the feature level where 1..N services provide a feature. Let's discuss that as part #194